shutdown.go

```diff
@@ -39,6 +39,18 @@ type ShutdownOption interface {
 	apply(*shutdowner)
 }
 
+type exitCodeOption int
+
+func (e exitCodeOption) apply(s *shutdowner) {
```

Comment:
let's make the option implement `String()` so that it can be printed when debugging.